File: BE-SPRINT2/seafood-be/src/main/java/com/example/seafoodbe/service/impl/ProductService.java
package com.example.seafoodbe.service.impl;

import com.example.seafoodbe.model.IProduct;
import com.example.seafoodbe.model.ISellingProducts;
import com.example.seafoodbe.model.Product;
import com.example.seafoodbe.repository.IProductRepository;
import com.example.seafoodbe.service.IProductService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class ProductService implements IProductService {
    @Autowired
    private IProductRepository productRepository;


    @Override
    public void addProduct(String name, double price, Integer categoryId, Integer originId) {
        productRepository.addProduct(name, price, categoryId, originId);
    }

    @Override
    public Page<IProduct> showList(String search, Pageable pageable) {
        return productRepository.showList(search, pageable);
    }


    @Override
    public IProduct findByIdProduct(Integer productId) {
        return productRepository.findByIdProduct(productId);
    }

    @Override
    public Optional<Product> findById(Integer id) {
        return productRepository.findById(id);
    }

    @Override
    public List<ISellingProducts> sellingProducts() {
        return productRepository.sellingProducts();
    }

    @Override
    public Page<Product> getAll(Pageable pageable) {
        return productRepository.findAll(pageable);
    }
}

File: FE-SPRINT2/seafood-fe/src/app/service/product.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Product} from "../model/product";

@Injectable({
  providedIn: 'root'
})
export class ProductService {
  API_PRODUCT = 'http://localhost:8080/api/product'

  constructor(private httpClient: HttpClient) { }

  getAll(search: any, size: any): Observable<any[]>{
   return  this.httpClient.get<any[]>(this.API_PRODUCT+'/list?search='+search+'&size='+size)
  }

  findById(productId: number){
    return this.httpClient.get<Product>(this.API_PRODUCT+ '/findById?productId='+productId)
  }

  sellingProducts(size: any): Observable<any[]>{
    return  this.httpClient.get<any[]>(this.API_PRODUCT+'/sellingProducts?size='+size)
  }

  showList(page: any): Observable<Product[]>{
    return  this.httpClient.get<Product[]>(this.API_PRODUCT+'/getAll?page='+page)
  }

}
